src/blueapi/service/main.py

```diff
@@ -214,9 +216,21 @@ def get_plan_by_name(name: str, runner: WorkerDispatcher = Depends(_runner)):
 
 @secure_router.get("/devices", response_model=DeviceResponse)
 @start_as_current_span(TRACER)
-def get_devices(runner: WorkerDispatcher = Depends(_runner)):
+def get_devices(
+    depth: Annotated[
+        int,
+        Query(
+            description="Maximum depth of children to return: "
+            "-1 for all, 0 for just root",
+            ge=-1,
+            # https://github.com/fastapi/fastapi/discussions/13473
+            json_schema_extra={"description": None},
```

Comment:
https://github.com/fastapi/fastapi/discussions/13473

I'll make an issue for tracking this if this is the way we want to go